src/util/http/HttpServer.h

```diff
@@ -216,13 +222,26 @@ class HttpServer {
         co_await http::async_read(stream, buffer, req,
                                   boost::asio::use_awaitable);
 
-        // Currently there is no timeout on the server side, this is handled
-        // by QLever's timeout mechanism.
-        stream.expires_never();
+        if (beast::websocket::is_upgrade(req)) {
+          auto errorResponse = ad_utility::websocket::checkPathIsValid(req);
```

Comment:
Not sure how you imagine the member function to look like. The main reason for this code construct is because of the conditional move in the if. If the websocket manager is not responsible nothing is done and nothing is moved anywhere. But if it is reponsible, the ansible cleanup is canceled and the request/socket stuff is moved to the manager.
Do you see a better approach to this?